src/FakeItEasy/Configuration/BuildableCallRule.cs

```diff
@@ -160,6 +160,24 @@ public virtual void ApplyWherePredicate(Func<IFakeObjectCall, bool> predicate, A
 
         public abstract void UsePredicateToValidateArguments(Func<ArgumentCollection, bool> predicate);
 
+        /// <summary>
+        /// Clones the part of the rule that describes which call is being configured.
+        /// </summary>
+        /// <returns>The cloned rule.</returns>
+        public BuildableCallRule CloneCallSpecification()
+        {
+            var clone = this.CloneCallSpecificationCore();
+            clone.wherePredicates.AddRange(this.wherePredicates);
+            return clone;
+        }
+
+        /// <summary>
+        /// When overridden in a derived class, returns a new instance of the same type and copies the call
+        /// specification members for that type.
+        /// </summary>
+        /// <returns>The cloned rule.</returns>
+        protected abstract BuildableCallRule CloneCallSpecificationCore();

```

Comment:
- Generic type param: how ? `BuildableCallRule` itself would need to become generic, using the [CRTP](https://en.wikipedia.org/wiki/Curiously_recurring_template_pattern). Is this what you had in mind?

        internal abstract class BuildableCallRule<T>
            : IFakeObjectCallRule
              where T : BuildableCallRule<T>
        {
            ...
            protected abstract T CloneCallSpecificationCore();
            ...
        }

        internal class AnyCallCallRule
            : BuildableCallRule<AnyCallRule>
        {
            ...
            protected override AnyCallRule CloneCallSpecificationCore()
            {
                return new AnyCallCallRule
                {
                    argumentsPredicate = this.argumentsPredicate
                };
            }
        }

    I think it's a little overkill...

- "core" suffix: it's a common naming convention when using the Template Method pattern (very frequent in the BCL). But if you have a better idea, I'm open to suggestions.